tools/params_processors/swift_debug_settings_processor.py

```diff
@@ -14,13 +14,128 @@ def _build_setting_path(path):
     return path
 
 
+def _is_relative_path(path: str) -> bool:
+    return not path.startswith("/") and not path.startswith("__BAZEL_")
+
+
+_CLANG_SEARCH_PATHS = {
+    "-iquote": None,
+    "-isystem": None,
+    "-I": None,
+}
+
 _ONCE_FLAGS = {
     "-D": None,
     "-F": None,
     "-I": None,
 }
 
 
+def _process_clang_opt(opt, previous_opt, previous_clang_opt):
+    if opt == "-Xcc":
+        return None
+    if previous_opt != "-Xcc":
+        return None
+
+    if opt.startswith("-F"):
+        path = opt[2:]
+        if path == ".":
+            return "-F$(PROJECT_DIR)"
+        if _is_relative_path(path):
+            return "-F$(PROJECT_DIR)/" + path
+        return opt
+    elif opt.startswith("-fmodule-map-file="):
+        path = opt[18:]
+        if path == ".":
+            return "-fmodule-map-file=$(PROJECT_DIR)"
+        if _is_relative_path(path):
+            return "-fmodule-map-file=$(PROJECT_DIR)/" + path
+        return opt
+    elif opt.startswith("-iquote"):
+        path = opt[7:]
```

Comment:
Nit: `path = opt[len("-iquote"):]` it will be more readable.

Reply:
We know the length though, so I'm going to opt out of it for now. I can add a comment though.